WcaOnRails/app/assets/javascripts/application.js

```diff
@@ -46,6 +47,60 @@ wca.cancelPendingAjaxAndAjax = function(id, options) {
   return wca._pendingAjaxById[id];
 };
 
+$.fn.competitionsMap = function(competitions) {
+  var $map = new google.maps.Map(document.getElementById(this.attr('id')), {
+    zoom: 2,
+    center: {lat: 0, lng: 0},
+    scrollwheel: true
+  });
+
+  var markers = [];
+
+  competitions.forEach(function(c) {
+
+    var contentString = "<a href=" + c.url + ">" + c.name + "</a><br />" + c.marker_date + " - " + c.cityName;
+
+    var infowindow = new google.maps.InfoWindow({
+      content: contentString
+    });
+
+    if (c.is_over) {
+      iconImage = 'http://maps.google.com/mapfiles/ms/icons/blue.png';
+    } else {
+      iconImage = 'http://maps.google.com/mapfiles/ms/icons/red.png';
+    }
+
+    c.marker = new google.maps.Marker({
+      map: $map,
+      position: {
+        lat: c.latitude_degrees,
+        lng: c.longitude_degrees,
+      },
+      title: c.name,
+      icon: iconImage
+    });
+
+    c.marker.addListener('click', function() {
+      infowindow.open($map, c.marker);
+    });
+
+    markers.push(c.marker);
+  });
+
+  var markerCluster = new MarkerClusterer($map, markers, {
+    maxZoom: 10,
+    clusterSize: 30
+  });
+};
+
+function detect_mobile() {
```

Comment:
If we're going to keep doing this (and I'm not convinced we should), we have to rename this method to `isMobile`, and you have to link to where this code came from.

Here are the constraints as I understand them:

- [ ] @mkoryak said we're not supposed to use floatThead on "mobile". As he's the author of floatThead, I'm not going to try to argue with that, surprising as it is to me. It's not clear to me if "mobile" in this context includes tablets, however. Most user agent string parsers seem to treat mobile differently than tablets, because they're usually doing this detection to decide how to lay things out on tiny screens, it's not used as a way to detect how powerful a CPU is.
- [ ] On small screens, we don't want the new competition form picker to float (for the record, we're not using floatThead here), simply because it will take up most of the screen.

I propose using [Responsive Bootstrap Toolkit](https://github.com/maciej-gurban/responsive-bootstrap-toolkit) to detect if we're on `sm` or larger, and only enabling the floating then. This will have a side effect of disabling floatThead on on mobile phones, and we don't have to parse the user agent string.